elasticsearch/elasticsearch.d.ts

```diff
@@ -331,6 +333,11 @@ declare module Elasticsearch {
         refresh?: boolean;
     }
 
+    export interface DeleteByQueryParams extends GenericParams {
+        index: string;
```

Comment:
They are not equal.

For instance deletebyquery has the following property: waitForActiveShards which search does not have.

They might have the same fields, but they are not the same in a "operational way". Reuse serves little purpose here,, and only bring tighter coupling where its not needed IMO.  This demands a new deletebyquereryparams... you can copy paste the identical properties between the 2.